src/screens/Platform/exchange/CompleteExchange.js

```diff
@@ -0,0 +1,128 @@
+import { useBroadcast } from "../../../components/useBroadcast";
+import DeviceActionModal from "../../../components/DeviceActionModal";
+
+type Result = {
+  operation?: Operation,
+  error?: Error,
+};
+
+export default function PlatformCompleteExchange({
+  route: {
+    params: { request, onResult, device },
+  },
+  navigation,
+}: {
+  navigation: *,
+  route: {
+    params: {
+      request: {
+        exchangeType: number,
+        provider: string,
+        exchange: {
+          fromAccount: Account,
+          fromParentAccount: Account,
+          toAccount: Account,
+          toParentAccount: Account,
+        },
+        transaction: Transaction,
+        binaryPayload: string,
+        signature: string,
+        feesStrategy: string,
+      },
+      device: Device,
+      onResult: (result: Result) => void,
+    },
+  },
+}) {
+  const {
+    fromAccount: account,
+    fromParentAccount: parentAccount,
+  } = request.exchange;
+  let tokenCurrency;
+  if (account.type === "TokenAccount") tokenCurrency = account.token;
+
+  const broadcast = useBroadcast({ account, parentAccount });
+  const [transaction, setTransaction] = useState();
+  const [signedOperation, setSignedOperation] = useState();
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    if (signedOperation) {
+      broadcast(signedOperation).then(operation => {
+        onResult({ operation });
+      }, setError);
+    }
+  }, [broadcast, onResult, signedOperation]);
+
+  useEffect(() => {
+    if (error) {
+      onResult({ error });
+    }
+  }, [onResult, error]);
+
+  return (
+    <SafeAreaView style={styles.root}>
+      {!transaction ? (
+        <DeviceActionModal
+          key="completeExchange"
+          device={device}
+          action={exchangeAction}
+          onClose={() => navigation.pop()}
+          onResult={({ completeExchangeResult, completeExchangeError }) => {
+            if (completeExchangeError) {
+              setError(completeExchangeError);
+            } else {
+              setTransaction(completeExchangeResult);
+            }
+          }}
+          request={request}
+        />
+      ) : (
+        <DeviceActionModal
+          key="sign"
+          device={device}
+          action={sendAction}
+          onClose={() => navigation.pop()}
+          onResult={({ signedOperation, transactionSignError }) => {
+            if (transactionSignError) {
+              setError(transactionSignError);
+            } else {
+              setSignedOperation(signedOperation);
+            }
+          }}
+          request={{
+            tokenCurrency,
+            parentAccount,
+            account,
+            transaction,
+            appName: "Exchange",
+          }}
+        />
+      )}
+    </SafeAreaView>
+  );
+}
+
+const exchangeAction = createAction(completeExchange);
+const sendAction = txCreateAction(connectApp);
+
+const styles = StyleSheet.create({
+  root: {
+    flex: 1,
+    padding: 32,
+  },
+});
```

Comment:
Couldn't we move the definition of these variables before the declaration of the `PlatformCompleteExchange` function (l 23) to follow the [no-use-before-define](https://eslint.org/docs/rules/no-use-before-define) rule and therefore reduce confusion?

Reply:
To me the most important part of the file is its props and functional component and these should be defined at the top most. It's the same reason as defining styles in the bottom. If you think `no-use-before-define` rule is the best bet, then it should be forced by a linter. Forcing without linter would become unmanageable eventually.